### ISSUE TITLE
multiple NPE's when  opening theme editor

### ISSUE THREAD

**Issue by childsb** [maqetta/maqetta#859]
0) enable halt on errors in javascript debugger (FF) 
1) create a new theme
2) (theme editor auto launches on new theme).  notice the error or the debugger stop with the message:
"davinci.Runtime.currentEditor.visualEditor._editorSelected is undefined"
3) Also note the "this._widgets[w] is undefined" error after continuing after error #2  (not sure if they are related).

**Comment by childsb** [comment]
Also noticed this error when closing the theme and visual editor: " davinci.Runtime.currentEditor is null "

**Comment by jhpedemonte** [comment]
@billreed63, can you take a look?

**Comment by jhpedemonte** [comment]
Actually, Brad says this might be due to Adam's changes to move the editor subscriptions out to Runtime.js.  Reassigning to @peller for now.

**Comment by peller** [comment]
@childsb, can you retest?  It seems ok now, and @billreed63 corrected my changes relative to ThemeEditor in 3ab9902e7346946bc711f7e0c00e2ca6662d9397

**Comment by peller** [comment]
Yeah, in davinci.ve.themeEditor.metadata._updateStyle, line 217, this._widgets[w] is undefined where w=="uniqName_1"  @billreed63, can you take a look?  @childsb, you said something about the model being wrong.  Can you elaborate here?

**Comment by billreed63** [comment]
The issue appears to be related to dijit.form.TextBox
When the theme editor is loading it calls  davinci.ve.themeEditor.metadata.metadata._updateStyle for each widget on the page. when the method is passed the TextBox widget the exception is thrown. The reason an exception is thrown is that the widget.type= uniqeName_## the widget.type is set from the dijit.declareType which is set by dojo when the widget is constructed. dojo/_base/declare.js line 31 

```
if(!proto.hasOwnProperty("declaredClass")){
    proto.declaredClass = "uniqName_" + (counter++);
}
```

This only occurs on FF, chrome works correctly...

Why is the declaredClass wrong for FF only? Maybe this will e fixed with the latest dojo 1.7 RC

**Comment by peller** [comment]
@billreed63

ok, the declared issue will be dealt with in #928.  Now, when I try to open the theme Editor, I get

e.widget.getParent is not a function

where e.widget is a string, "$all".  I see this on Chrome, or on FF where I've worked around #928.

**Comment by billreed63** [comment]
fixed by #906
